### DISCUSSION TITLE
Difference between opentelemetry-spring-boot-starter and the OpenTelemetry support in micrometer

### DISCUSSION THREAD

**Discussion by silver-mx** [open-telemetry/opentelemetry-java-instrumentation#11403] [Q&A] [answered]
I am trying to find out the easiest and most reliable way to enable OpenTelemetry observability in SpringBoot. On one hand we have the OpenTelemetry micrometer support for [metrics][1] and [tracing][2]. Tracing requires an exporter provided by OTEL as well:

```
    <dependency>
        <groupId>io.opentelemetry</groupId>
        <artifactId>opentelemetry-exporter-otlp</artifactId>
    </dependency>
```

On the other hand, we have the official [OpenTelemetry starter dependency `opentelemetry-spring-boot-starter`][3] which does autoconfiguration, instrumentation and brings several additional dependencies not included by default in SpringBoot. 

To the best of my knowledge there is some overlapping in both solutions as they do instrumentation to some degree, but I would like to know the following:

 1. What offers `opentelemetry-spring-boot-starter` that micrometer + `opentelemetry-exporter-otlp` does not already have.
 2. In which scenarios I would choose `opentelemetry-spring-boot-starter` instead of the micrometer support.

__NOTE:__ I posted the same question in [StackOverflow ](https://stackoverflow.com/questions/78185793/difference-between-opentelemetry-spring-boot-starter-and-the-opentelemetry-suppo)but nobody has answered it, so I thought trying here.

  [1]: https://docs.micrometer.io/micrometer/reference/implementations/otlp.html
  [2]: https://docs.micrometer.io/tracing/reference/tracers.html
  [3]: https://opentelemetry.io/docs/languages/java/automatic/spring-boot/#opentelemetry-starter-dependency

**Comment by trask** [comment, marked as answer]
hi @silver-mx!

At a very very high level:

OpenTelemetry Spring Boot starter
* Natively uses the OpenTelemetry API, OpenTelemetry SDK and the OpenTelemetry semantic conventions
* Supported by the OpenTelemetry community (governed by the CNCF / Linux Foundation)

Micrometer
* An independent standalone observability system
* Offers optional bridges into parts of the OpenTelemetry ecosystem
* Supported by the Micrometer community (governed by Broadcom Inc)

As a member of the OpenTelemetry Java community, I (probably unsurprisingly) recommend using the OpenTelemetry Spring Boot starter. We have a great community behind it, and are continuously making improvements. If you try it out and have any problems or suggestions, please open an [issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues) in our repository.

**Comment by lucian-vanghele** [comment]
isn't javaagent the third option?

**Reply by silver-mx** [reply]
@lucian-vanghele it could probably be, but my question was originally focused on the spring-boot integration given either by micrometer or opentelmetry, while the agent I assume is more a generic solution for any Java application right? I work mostly with Springboot applications so I limited the question to that.

**Comment by DominicBortmes** [comment]
Hi all this was so helpful and made my day, thanks!
I noticed a big naming overlap that makes it hard to navigate the different options/libraries:

Both of the above approaches use java libraries that have the almost exact same names, containing`spring-boot-starter` and `opentelemetry` substrings. The libraries seem to vary in implementation details (properties, otel sdk env var support, etc: example aggregationTemporality vs aggregation.temporality) and have distinct docs. 

So this call out is to explain to everyone that:

Option "cncf otel impl." 1 as per above answer is from groupId `io.opentelemetry.instrumentation`:
  ```
<dependency>
      <groupId>io.opentelemetry.instrumentation</groupId>
      <artifactId>opentelemetry-**spring-boot-starter**</artifactId>
  </dependency>

```
Option 2 "micrometer" as per above answer is from groupId `org.springframework.boot` - also called a spring boot starter.
```
    <groupId>org.springframework.boot</groupId>
    <artifactId>**spring-boot-starter**-opentelemetry</artifactId>
```

Maybe someone else out there who is an SME on the topic can clarify further, but thread carefully!
